crates/sui-types/src/temporary_store.rs

```diff
@@ -883,13 +878,10 @@ impl<S: ObjectStore> TemporaryStore<S> {
                         )
                     } else {
                         // else, this is an input object, not a dynamic field
-                        if let Ok(Some(old_obj)) = self.store.get_object(object_id) {
-                            let expected_version = object.version();
-                            if old_obj.version() != expected_version {
-                                return Err(ExecutionError::invariant_violation(
-                                    "Expected to find old object with version {expected_version}",
-                                ));
-                            }
+                        let expected_version = object.version();
+                        if let Ok(Some(old_obj)) =
+                            self.store.get_object_by_key(object_id, expected_version)
```

Comment:
I think here as well this should exist? Should we panic if we do not find the right object?

Reply:
Agreed, same here on keeping invariant violation.

Reply:
reverted this code given it is legacy mode and we want to leave it untouched